core/dbt/config/project.py

```diff
@@ -144,44 +150,61 @@ def _parse_versions(versions):
     return [VersionSpecifier.from_version_string(v) for v in versions]
 
 
+def _all_source_paths(
```

Comment:
why enumerate the different resource paths here instead of just passing in a list of paths, or varargs, or something like that? I buy it, just curious about your reasoning

Reply:
My reasoning was that at some point in the future, we're going to have to update this (macros and analyses). Currently two different places call this function and the data should be in sync.

With this signature, either mypy/pylint will catch it or we'll crash at runtime with a line pointing to the call that's missing arguments during parsing. That will never accidentally get merged because no tests will ever pass. With a list of paths/varargs, it'll be easy to forget one place or the other and not notice.

Reply:
awesome, makes a ton of sense